### PR TITLE
Add support for MultiColumnField to support multiline

### PR DESCRIPTION
**Pull request by andrewballantyne**
**Fixes**: 
https://issues.redhat.com/browse/ODC-5742

**Analysis / Root cause**: 
MultiColumnField assumed everything is a singular line height - this was problematic for the design in Pipeline Builder.

**Solution Description**: 
Supported an optional complex field setting to avoid over reaching into fields and assuming their size / layout.

**Screen shots / Gifs for design review**: 

Pipeline Builder Example...

Single entry:
Screen Shot 2021-04-06 at 3 28 45 PM

Multiple entries:
image

Multiple `MultiColumnField`s:
Screen Shot 2021-04-06 at 3 28 19 PM

**Unit test coverage report**: 
No change.

**Test setup:**

Go to Pipeline Builder and add workspaces.

**Browser conformance**: 
- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by andrewballantyne** [comment]
/hold

Some dispute on layout/design... Waiting on @bgliwa01's approval before this can merge. Any reviews can be conducted against the code design, little of the design changes will impact the overall coding result.

File: frontend/packages/console-shared/src/components/formik-fields/multi-column-field/MultiColumnFieldRow.tsx
Comment (abhinandan13jan) [inline frontend/packages/console-shared/src/components/formik-fields/multi-column-field/MultiColumnFieldRow.tsx:37]: Why do we need to have an array and store value for every column? We could simply have this flag at the row level. WDYT?

File: frontend/packages/pipelines-plugin/src/components/pipelines/detail-page-tabs/OptionalableWorkspace.tsx
Comment (abhinandan13jan) [inline frontend/packages/pipelines-plugin/src/components/pipelines/detail-page-tabs/OptionalableWorkspace.tsx:25]: Optional workspace

File: frontend/packages/pipelines-plugin/locales/en/pipelines-plugin.json
Comment (abhinandan13jan) [inline frontend/packages/pipelines-plugin/locales/en/pipelines-plugin.json:52]: workspace

File: frontend/packages/console-shared/src/components/formik-fields/multi-column-field/MultiColumnFieldRow.tsx
Comment (abhinandan13jan) [inline frontend/packages/console-shared/src/components/formik-fields/multi-column-field/MultiColumnFieldRow.tsx:43]: Should we use a better check than `!complexFields`?

Comment (rohitkrai03) [reply frontend/packages/console-shared/src/components/formik-fields/multi-column-field/MultiColumnFieldRow.tsx:43]: Agreed. This will always result in `false`. Maybe you need to use `_.isEmpty(complexFields)` here.

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]
Overall changes looks good to me. There's just one comment.

**Comment by andrewballantyne** [comment]
Just to answer your question @abhinandan13jan - you deleted it but to clarify for a learning experience here.

> Why do we need to have an array and store value for every column? We could simply have this flag at the row level. WDYT?

We don't want to store it as a singular variable because you may mix regular Formik Fields with a "complex" one. If it's complex, it doesn't dig into the props looking for `name` to update the Formik name prop. So there may be cases where we want both to happen at the same time.

File: frontend/packages/console-shared/src/components/formik-fields/multi-column-field/MultiColumnFieldRow.tsx
Comment (andrewballantyne) [reply frontend/packages/console-shared/src/components/formik-fields/multi-column-field/MultiColumnFieldRow.tsx:43]: Good point, thanks guys! Easy update.

**Comment by andrewballantyne** [comment]
/unhold

Brigid is checking with UX, but there is no need to hold up this PR on that conversation. We will address it via another PR if there are changes to be made.

**Review by bgliwa01 — APPROVED** [review]
I will confirm with UX but approving for now!

**Review by rohitkrai03 — APPROVED** [review]
/lgtm

**Pull request merged**
